src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -118,6 +119,54 @@ public override async Task<dynamic> GetSite(string stampName, string siteName)
             return siteObject;
         }
 
+        public override async Task<string> GetStampName(string subscriptionId, string resourceGroupName, string siteName)
+        {
+            dynamic siteObjects = await GetSite(siteName);
+            JToken obj2 = ((JArray)siteObjects)
+                    .Select(i => (JObject)i)
+                    .FirstOrDefault(
+                        j => (j.ContainsKey("Subscription") && j["Subscription"].ToString() == subscriptionId
+                            && j.ContainsKey("ResourceGroupName") && j["ResourceGroupName"].ToString() == resourceGroupName
+                            && j.ContainsKey("stampName")));
+
+            string stampName = obj2?["stampName"]?.ToString();
+            return stampName;
+        }
+
+        public override async Task<dynamic> GetHostNames(string siteName)
+        {
+            var response = await Get($"sites/{siteName}/hostnames");
```

Comment:
>$"sites/{siteName}/hostnames") [](start = 37, length = 30)

Use this path: stamps/{stamp}/sites/{site}/hostnames.
SiteNames are not unique so sometimes we can have a sitename in a public stamp and that same sitename in a ministamp. This call will return hostnames for both sites.

Reply:
Oh I see, Good Point. Let me change this.